contracts/REXReferral.sol

```diff
@@ -55,6 +56,22 @@ contract REXReferral is Ownable {
         _;
     }
 
+    /// @dev Check if addr in allowedUsers
+    modifier onlyAllowedAddr() {
+        require(allowedUsers[msg.sender], "Access denied. Not eligible to call this function.");
+        _;
+    }
+
+    // Whitelist an address
+    function whitelistAddress (address userAddr) public onlyOwner {
+        allowedUsers[userAddr]=true;
+    }
+
+    // Remove from whitelist
+    function removeFromWhitelistAddress (address userAddr) public onlyOwner {
+        allowedUsers[userAddr]=false;
+    } 
+
```

Comment:
Could you use Open zepplien `Roles.sol` like are used in REX Bank, like this PR shows: https://github.com/Ricochet-Exchange/rex-bank/pull/31

Roles has all the functionality implemented for adding/removing

There should be two roles: `Admin` and `ApprovedApp`

Also, whitelist should be "allowlist", but I think it you are using OZ Roles then you wont need to code these yourself

Reply:
Done 👍